content/blog/2020/release-0.2.0.md

```diff
@@ -0,0 +1,190 @@
+places.
+
+This release being particularly large contains a few different API-breaking changes: going forward we expect to make the interface more and more stable, which in turn will make applying updates easier.
+
+### Branch and Bound coin selection
+
+We now support the state-of-the-art coin selection algorithm called "branch and bound", with an implementation derived straight from Bitcoin Core. This algorithm is now enabled by default, but it can be replaced with a different one (either
+the old default, [`LargestFirstCoinSelection`][`LargestFirstCoinSelection`] or a custom [`CoinSelectionAlgorithm`][`CoinSelectionAlgorithm`]) by using the [`TxBuilder::coin_selection()`][`TxBuilder::coin_selection()`] option.
+
+Branch and bound works by trying to find a set of inputs that perfectly matches the amount being sent by a transaction, to avoid making an extra change output which takes up more space in the transaction, requires more fees, and in general lowers the privacy
+of a user if the change is later spent together with other outputs.
+
+### Key generation
+
+If you need to generate a new `bip32::ExtendedPrivKey`, or perhaps a new BIP39 mnemonic, you can use the unified [`GeneratableKey`][`GeneratableKey`] trait to do so: paired with [`GeneratableDefaultOptions`][`GeneratableDefaultOptions`] they provide many different ways to generate keys,
+with or without a custom source of entropy, and with or without customized options.
+
+```rust
+use bdk::bitcoin::PrivateKey;
+use bdk::keys::{GeneratableKey, GeneratableDefaultOptions, PrivateKeyGenerateOptions};
+
+let default_options_key = PrivateKey::generate_default()?;
+let custom_options_key = PrivateKey::generate(PrivateKeyGenerateOptions { compressed: false })?;
+```
+
+### Generic key types
+
+With this update there's now a generalized trait for keys that can be used in descriptors, which is called [`ToDescriptorKey`][`ToDescriptorKey`]. This trait is already implemented for the native `rust-bitcoin` key types, like `PrivateKey`, `PublicKey`, `bip32::ExtendedPrivKey`
+and `bip32::ExtendedPubKey`. It's also implemented for BIP39 mnemonic and seeds, when the the opt-in `keys-bip39` feature is enabled. As always, being this a public trait, you can also implement it for custom types to better suit your needs.
+
+```rust
+impl<Ctx: ScriptContext> ToDescriptorKey<Ctx> for MyKeyType {
+    fn to_descriptor_key(self) -> Result<DescriptorKey<Ctx>, KeyError> {
+        // Custom conversion to `bitcoin::PrivateKey`
+        let privkey: bitcoin::PrivateKey = ... ;
+        privkey.to_descriptor_key()
+    }
+}
+```
+
+If your custom key types is simply a different representation of an `xprv` or `xpub`, you can also consider implementing the [`DerivableKey`][`DerivableKey`] trait instead: for a type `K` that implements [`DerivableKey`][`DerivableKey`], the [`ToDescriptorKey`][`ToDescriptorKey`] trait is automatically
```

Comment:
just noticed this.. I believe should be singular: "If your custom key type is simply..."